bundles/framework/divmanazer/component/FileInput.js

```diff
@@ -99,27 +214,74 @@ Oskari.clazz.define('Oskari.userinterface.component.FileInput', function () {
                         cb( fileContent );
                     };
                 })(f);
-
                 reader.readAsText(f);
             }
         },
         /**
          * @method create
          * Creates the element for handlign drag and drop
          */
-        create: function() {
-            var fileinput = this._template.fileBox({ link: this.loc('fileInput').link,
-                                                    fileupload: this.loc('fileInput').fileupload,
-                                                    uploading: this.loc('fileInput').uploading,
-                                                    success: this.loc('fileInput').success,
-                                                    error: this.loc('fileInput').error });
-            this.setElement(fileinput);
+        createUi: function() {
+            var allowedFiles = this.getAcceptedTypesString();
+            var classes;
+            var fileInput;
+            var fileUpload;
+            var opts = this.options;
+            var allowMultiple;
 
-            // if( this.canUseAdvancedUpload ) {
-            //     this.handleDragAndDrop( fileinput );
-            // }
-            // return this.getElement();
+            if (opts.allowMultipleFiles === true ){
+                allowMultiple = "multiple";
+                fileUpload = this.loc('fileInput.fileUpload', {files: 2});
+            } else {
+                allowMultiple = "";
+                fileUpload = this.loc('fileInput.fileUpload', {files: 1});
+            }
+
+            if (this.isAdvancedUpload){
+                classes = "oskari-fileinput advanced-upload";
```

Comment:
I don't think classes need to be in a variable. You can just use the string when calling the template-function.